app/services/service_provider_updater.rb

```diff
@@ -1,7 +1,13 @@
 class ServiceProviderUpdater
+  # :reek:TooManyStatements
   def self.ping
     resp = HTTParty.post(idp_url, headers: token_header)
-    resp.code == 200
+    status_code = resp.code
+    return status_code if status_code == 200
+
+    handle_error(status_code)
+  rescue StandardError => error
+    handle_error(error.msg)
```

Comment:
there's probably a better way to write this? 😴

Reply:
@amoose check out my updates...